utils/sorting.go

```diff
@@ -20,24 +19,29 @@ type Sortable[T any] interface {
 
 // Sorts the elements of [s].
 func Sort[T Sortable[T]](s []T) {
-	slices.SortFunc(s, T.Less)
+	slices.SortFunc(s, func(a, b T) int {
+		if a.Less(b) {
+			return -1
+		}
+		return 1
```

Comment:
I don't think this correctly handles the case that `a` is equal to `b`

Reply:
Can we use `sort.Slice` instead?
```go
func Slice(x any, less func(i, j int) bool)
```
https://pkg.go.dev/sort#Slice

Reply:
`sort.Slice` uses reflection and is slower than `slices.SortFunc` / `slices.Sort`

Reply:
I think that `sort.Interface` could be implemented if the reflection is really a concern :shrug:

Reply:
Personally, I like option 3 the best. Something like this seems reasonable:

```go
type Sortable[T any] interface {
	Compare(T) int
}
```

Reply:
I had originally thought we would just do:
```
if a.Less(b) {
	return -1
}
if b.Less(a) {
	return 1
}
return 0
```
But changing the `Sortable` interface is a better long term strategy imo.